### PR TITLE
feat(control-panel): add registry api interface and repository

### PR DESCRIPTION
**Pull request by keplervital**
This PR adds the registry api interface and repositories for the registry. I've only included some of the get implementation in the service layer for now and had to move the pagination from the station canister to the common orbit_essentials lib to be reused by the control-panel. 

Also, i'll still create a follow up PR that will remove the pagination implementation from the station canister and will start to use it from the common orbit_essentials lib.

### CODE REVIEW

File: core/control-panel/api/spec.did [hunk 1]
```diff
@@ -310,8 +319,254 @@ type GetArtifactResult = variant {
...
 };
 
+// A metadata record that contains a key and a value.
+type Metadata = record {
+  key : text;
+  value : text;
+};
+
+// An entry record, which contains information and the value stored in the registry.
+type RegistryEntry = record {
```
Comment (olaszakos) [inline core/control-panel/api/spec.did:329]: These were renamed in Rust, intended to for Candid?

File: core/control-panel/api/spec.did [hunk 2]
```diff
@@ -310,8 +319,254 @@ type GetArtifactResult = variant {
...
+  //
+  // e.g. if the namespace is "@orbit" and the name is "station", then all the entries will refer to a wasm module.
+  //
+  // Restrictions:
+  //
+  // - Names that start with `@` are considered namespaced.
+  // - Names that start with `@` must have a namespace and a name separated by a `/`.
+  // - Names must be between 2 and 48 characters long.
+  // - Namespaces must be between 2 and 32 characters long.
+  // - Names that are not namespaced, are put in the default namespace `@a`.
```
Comment (olaszakos) [inline core/control-panel/api/spec.did:344]: default namespace is `@default`, same for the comment in the Rust counterpart

File: core/control-panel/impl/src/models/registry.rs [hunk 3]
```diff
@@ -12,21 +12,18 @@ use std::collections::BTreeMap;
...
-/// The registry entry is a record that is stored in the registry repository.
+/// The registry is a record that is stored in the registry repository.
 ///
 /// It stores entries about wasm modules and can be extended to other entry types. When adding new entry types,
-/// the `RegistryEntryValue` enum should be updated to include the new entry type.
+/// the `RegistryValue` enum should be updated to include the new entry type.
 #[storable]
 #[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
-pub struct RegistryEntry {
+pub struct Registry {
```
Comment (olaszakos) [inline core/control-panel/impl/src/models/registry.rs:24]: To me Registry sounds like a collection of items, not the item itself in the collection.

**Review by olaszakos — APPROVED** [review]

File: core/control-panel/impl/src/models/registry.rs (hunk 3 above)
Comment (keplervital) [reply core/control-panel/impl/src/models/registry.rs:24]: Makes sense, i'll rename this back to `RegistryEntry`

File: core/control-panel/api/spec.did (hunk 2 above)
Comment (keplervital) [reply core/control-panel/api/spec.did:344]: good catch

**Pull request merged**